### PR TITLE
feat: add `OnyxUserMenu` component

### PR DESCRIPTION
**Pull request by larsrickert**
Relates to #881

Add basic `OnyxUserMenu` component

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxListboxOption/OnyxListboxOption.vue [hunk 1]
```diff
@@ -39,8 +52,13 @@ defineSlots<{
   @include checkbox.variables();
 
   @include layers.component() {
+    --onyx-listbox-option-color: var(--onyx-color-text-icons-neutral-intense);
+    --onyx-listbox-option-color-active: var(--onyx-color-text-icons-primary-intense);
```
Comment (MajaZarkova) [inline packages/sit-onyx/src/components/OnyxListboxOption/OnyxListboxOption.vue:56]: Should we rename this to --onyx-listbox-options-color-selected, so it's aligned with the background-selected?

**Comment by MajaZarkova** [comment]
On the first click the listbox opens, but on second click on the button the listbox stays open. It closes only on outside click. Is this intentional?

**Review by MajaZarkova — CHANGES_REQUESTED** [review]

**Comment by MajaZarkova** [comment]
Also, should we close the listbox on option click as well?

File: packages/sit-onyx/src/components/OnyxListboxOption/OnyxListboxOption.vue (hunk 1 above)
Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxListboxOption/OnyxListboxOption.vue:56]: The color is only applied when the option is active, right? Not when its only selected?

**Comment by larsrickert** [comment]
> On the first click the listbox opens, but on second click on the button the listbox stays open. It closes only on outside click. Is this intentional?

I checked with Jannick, the menu should only close when clicking an option.
Is it fine for you when we do this in a follow up PR?

File: packages/sit-onyx/src/components/OnyxListboxOption/OnyxListboxOption.vue (hunk 1 above)
Comment (MajaZarkova) [reply packages/sit-onyx/src/components/OnyxListboxOption/OnyxListboxOption.vue:56]: I think it's applied when it's selected and active

**Comment by MajaZarkova** [comment]
> > On the first click the listbox opens, but on second click on the button the listbox stays open. It closes only on outside click. Is this intentional?
> 
> I checked with Jannick, the menu should only close when clicking an option. Is it fine for you when we do this in a follow up PR?

Sure, no problem

**Review by MajaZarkova — DISMISSED** [review]

**Review by MajaZarkova — APPROVED** [review]

**Pull request merged**